interfaces/src/dotty/tools/dotc/interfaces/SourcePosition.scala

```diff
@@ -0,0 +1,55 @@
+package dotty.tools.dotc.interfaces
+
+import scala.annotation.binaryCompatible
+
+
+/** A position in a source file.
+ *
+ * A position is a range between a start offset and an end offset, as well as a
+ * point inside this range.
+ *
+ * As a convenience, we also provide methods that return the line and the column
+ * corresponding to each offset.
+ *
+ * User code should not implement this interface, but it may have to
+ * manipulate objects of this type.
+ */
+@binaryCompatible
+trait SourcePosition {
+  /** @return Content of the line which contains the point */
+  def lineContent(): String
```

Comment:
Since all of these methods are pure, I think it's fine to rewrite them without any parameter lists when converting to Scala.